code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/NotificationBuilder.kt

```diff
@@ -67,9 +83,175 @@ object NotificationBuilder {
+        // at the later time
+        val remindLaterTimestamp =
+            intentExtras.getString(PushTemplateConstants.PushPayloadKeys.REMIND_LATER_TIMESTAMP)?.toLongOrNull() ?: 0
+        val remindLaterDuration =
+            intentExtras.getString(PushTemplateConstants.PushPayloadKeys.REMIND_LATER_DURATION)?.toLongOrNull() ?: 0
+
+        if (remindLaterTimestamp <= 0 && remindLaterDuration <= 0) {
+            Log.trace(
+                LOG_TAG,
+                SELF_TAG,
+                "Remind later timestamp or duration is less than or equal to current timestamp" +
+                    "will not schedule a notification from the received intent with action %s",
+                remindLaterIntent.action
+            )
+            return
+        }
+        val calendar: Calendar = Calendar.getInstance()
+        val notificationManager = NotificationManagerCompat.from(context)
+
+        // get the tag from the intent extras
+        val tag = intentExtras.getString(PushTemplateConstants.PushPayloadKeys.TAG)
+
+        // calculate difference in fire date from the current date if timestamp is provided
+        val secondsUntilFireDate: Long = if (remindLaterDuration > 0) remindLaterDuration
+        else remindLaterTimestamp - calendar.timeInMillis / 1000
+
+        // if fire date is greater than 0 then we want to schedule a reminder notification.
+        if (secondsUntilFireDate <= 0) {
+            Log.trace(
+                LOG_TAG,
+                SELF_TAG,
+                "Remind later date is before the current date. Will not reschedule the" +
+                    " notification.",
+                secondsUntilFireDate
+            )
+            // cancel the displayed notification
+            notificationManager.cancel(tag.hashCode())
+            return
+        }
+        Log.trace(
+            LOG_TAG,
+            SELF_TAG,
+            "Remind later pressed, will reschedule the notification to be displayed %d" +
+                " seconds from now",
+            secondsUntilFireDate
+        )
+        calendar.add(Calendar.SECOND, secondsUntilFireDate.toInt())
+
+        // schedule a pending intent to be broadcast at the specified timestamp
+        val scheduledIntent = Intent(PushTemplateConstants.IntentActions.SCHEDULED_NOTIFICATION_BROADCAST)
+        broadcastReceiverClass.let {
+            scheduledIntent.setClass(context.applicationContext, broadcastReceiverClass)
+        }
+        scheduledIntent.setFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP)
+        scheduledIntent.putExtras(intentExtras)
+
+        val pendingIntent: PendingIntent = PendingIntent.getBroadcast(
+            context,
+            0,
+            scheduledIntent,
+            PendingIntent.FLAG_MUTABLE or PendingIntent.FLAG_UPDATE_CURRENT
+        )
+
+        val alarmManager = context.getSystemService(Context.ALARM_SERVICE) as AlarmManager?
+        if (alarmManager != null) {
```

Comment:
Can you change this to 

```
  if ( alarmManager == null) {
    // log
    return;
  } 

  ...
   
```

Reply:
Changed and extracted the logic to common `PendingIntentUtils.createPendingIntentForScheduledNotifications`. It can it is also used by Timer Notification
cc: @PravinPK